ir/type.cpp

```diff
@@ -222,27 +252,60 @@ expr FloatType::getDummyValue() const {
 }
 
 expr FloatType::getTypeConstraints() const {
-  // TODO
-  return false;
+  if (defined)
+    return true;
+
+  auto bw = sizeVar();
+  // TODO: support more fp types
+  auto isFloat = bw == expr::mkUInt(Float, var_bw_bits);
+  auto isDouble = bw == expr::mkUInt(Double, var_bw_bits);
+  return (isFloat || isDouble);
 }
 
 expr FloatType::operator==(const FloatType &rhs) const {
-  // TODO
-  return false;
+  return sizeVar() == rhs.sizeVar();
 }
 
 expr FloatType::sameType(const FloatType &rhs) const {
   return true;
 }
 
 void FloatType::fixup(const Model &m) {
-  // TODO
+  if (!defined)
+    fpType = (FpType) m.getUInt(sizeVar());
+  assert(fpType >= Quarter && fpType <= FP128);
```

Comment:
you need to do the assert with an auxiliary unsigned var. Once you cast to FpType, it's UB if it falls outside the enum, so the assert will just go away.  See e.g. `SymbolicType::fixup()`.

Reply:
Oh, its interesting to know!